src/libos/src/fs/file_ops/read.rs

```diff
@@ -32,3 +32,20 @@ pub async fn do_pread(fd: FileDesc, buf: &mut [u8], offset: off_t) -> Result<usi
         file_ref.read(buf).await
     }
 }
+
+pub async fn do_preadv(fd: FileDesc, bufs: &mut [&mut [u8]], offset: off_t) -> Result<usize> {
+    debug!("preadv: fd: {}, offset {}", fd, offset);
+    if offset < 0 {
+        return_errno!(EINVAL, "the offset is negative");
+    }
+    let file_ref = current!().file(fd)?;
+    if let Some(async_file_handle) = file_ref.as_async_file_handle() {
+        if !async_file_handle.access_mode().readable() {
```

Comment:
```
        if !async_file_handle.access_mode().readable() {
            return_errno!(EBADF, "file is not readable");
        }
```

This code is useless because you have checked this inside `async_file_handle`